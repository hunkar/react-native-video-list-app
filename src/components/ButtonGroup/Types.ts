import {IconNode} from 'react-native-elements/dist/icons/Icon';
import {StyleProp, ViewStyle} from 'react-native';

//TODO Other styles can be customized. Can be added in the future.
/**
 * title: text of button
 * icon: icon of button
 * onPress: press event of button
 */
export type ButtonProps = {
  title: string;
  icon?: IconNode;
  onPress?: (() => void) | undefined;
};

/**
 * containerStyle: Styles for group container
 * buttons: array of ButtonProps
 */
export type ButtonGroupProps = {
  containerStyle?: StyleProp<ViewStyle>;
  buttons: ButtonProps[];
};
