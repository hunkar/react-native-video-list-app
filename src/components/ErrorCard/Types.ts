/**
 * title: Text of error.
 * onRetry: function that will be called on retry button clicked.
 */
export type ErrorCardProps = {
  title: string;
  onRetry?: () => void;
};
