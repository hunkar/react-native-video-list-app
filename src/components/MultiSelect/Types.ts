/**
 * id: id of item,
 * name: item name that will be shown
 * isSelected: you can set select status of item
 * value: extra value for item. For external use.
 */
export type MultiSelectItemProps = {
  id: string;
  name: string;
  isSelected?: boolean;
  value?: string | number | boolean;
};

/**
 * items: array of MultiSelectItemProps
 * selectedItems: id array of selected items
 * onChange: trigger function with selected items
 */
export type MultiSelectProps = {
  items: MultiSelectItemProps[];
  selectedItems?: string[];
  onChange?: (item: MultiSelectItemProps[]) => void;
};
