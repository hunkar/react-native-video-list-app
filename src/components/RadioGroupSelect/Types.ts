/**
 * id: item id
 * name: item name that will be show on the list
 * isSelected: item can be selectable with this prop
 * value: any value for external use.
 */
export type RadioGroupSelectItemProps = {
  id: string;
  name: string;
  isSelected?: boolean;
  value?: string | number | boolean;
};

/**
 * items: array of RadioGroupSelectItemProps items
 * selectedItem: id of selected item
 * onChange: trigger with selected item on change
 */
export type RadioGroupSelectProps = {
  items: RadioGroupSelectItemProps[];
  selectedItem?: string;
  onChange?: (item: RadioGroupSelectItemProps) => void;
};
