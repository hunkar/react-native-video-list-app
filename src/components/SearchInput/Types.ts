/**
 * placeholder: placeholder for searchBar
 * onChangeText: event for text change
 * changeTimeout: letter input frequency waiting range in ms 
 */
export type SearchInputProps = {
  placeholder?: string;
  onChangeText?: (text: string) => void;
  changeTimeout?: number;
};
