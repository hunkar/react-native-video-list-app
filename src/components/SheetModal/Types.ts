/**
 * title: title of modal
 * children: any ReactNode for modal child
 */
export type SheetModalProps = {
  title: string;
  children: React.ReactNode;
};
