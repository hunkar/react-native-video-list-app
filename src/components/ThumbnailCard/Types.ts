/**
 * id: Id of music.
 * title: Main title for singer name.
 * subTitle: Sub title for music name.
 * imageUrl: Url of video image.
 */
export type ThumbnailCardProps = {
  id: string;
  title: string;
  subTitle: string;
  imageUrl: string;
};
