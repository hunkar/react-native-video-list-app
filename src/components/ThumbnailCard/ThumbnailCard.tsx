import React, {useState} from 'react';
import {ImageSourcePropType, StyleSheet} from 'react-native';
import {Card, Text} from 'react-native-elements';
import {ThumbnailCardProps} from './Types';

/**
 * Default image url if uploading any video image is failed.
 */
const defaultImageUrl = 'https://static.thenounproject.com/png/2932881-200.png';

/**
 * 
 * Thumbnail card for music items.
 * 
 * @param props ThumbnailCardProps
 * @returns ReactNode
 */
const ThumbnailCard = (props: ThumbnailCardProps) => {
  const {title, subTitle} = props;
  const [imageUrl, setImageUrl] = useState<string>(props.imageUrl);

  return (
    <Card containerStyle={styles.card}>
      <Card.Image
        testID="image"
        style={styles.image}
        onError={() => {
          setImageUrl(defaultImageUrl);
        }}
        source={
          {
            uri: imageUrl,
          } as ImageSourcePropType
        }
      />
      <Card.Title testID="title" style={styles.title}>
        {title}
      </Card.Title>
      <Text testID="subTitle" style={styles.subTitle}>
        {subTitle}
      </Text>
    </Card>
  );
};

const styles = StyleSheet.create({
  card: {
    margin: 5,
    padding: 0,
    paddingBottom: 10,
  },
  image: {padding: 0},
  title: {marginTop: 25, marginBottom: 0},
  subTitle: {
    width: '100%',
    textAlign: 'center',
  },
});

export default ThumbnailCard;
