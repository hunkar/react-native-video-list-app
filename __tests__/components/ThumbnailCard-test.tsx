import renderer, {act} from 'react-test-renderer';
import React from 'react';

import ThumbnailCard from '../../src/components/ThumbnailCard/ThumbnailCard';
import {ThumbnailCardProps} from '../../src/components/ThumbnailCard/Types';

test('snapshot equility', async () => {
  const mockProps: ThumbnailCardProps = {
    id: '1',
    title: 'John',
    subTitle: 'Up',
    imageUrl: 'www.dummy-image.com/image.png',
  };

  const component = renderer.create(<ThumbnailCard {...mockProps} />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('renders areas correctly', async () => {
  const mockProps: ThumbnailCardProps = {
    id: '1',
    title: 'John',
    subTitle: 'Up',
    imageUrl: 'www.dummy-image.com/image.png',
  };

  const component = renderer.create(<ThumbnailCard {...mockProps} />);

  const titleComponent = component.root.findByProps({testID: 'title'}).props;
  const subTitleComponent = component.root.findByProps({
    testID: 'subTitle',
  }).props;
  const imageComponent = component.root.findByProps({testID: 'image'}).props;

  expect(titleComponent.children).toEqual(mockProps.title);
  expect(subTitleComponent.children).toEqual(mockProps.subTitle);
  expect(imageComponent.source.uri).toEqual(mockProps.imageUrl);
});
