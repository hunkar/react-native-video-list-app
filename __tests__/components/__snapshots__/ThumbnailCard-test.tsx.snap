// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot equility 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 5,
      "marginBottom": 0,
      "padding": 0,
      "paddingBottom": 10,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onError={[Function]}
        onLoad={[Function]}
        source={
          Object {
            "uri": "www.dummy-image.com/image.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 150,
            "left": 0,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": null,
          }
        }
        testID="image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        collapsable={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 150,
              "justifyContent": "center",
              "padding": 0,
              "width": null,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "height": 150,
            "padding": 0,
            "width": null,
          }
        }
        testID="RNE__Image__children__container"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#43484d",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 0,
          "marginTop": 25,
          "textAlign": "center",
        }
      }
      testID="title"
    >
      John
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
          "textAlign": "center",
          "width": "100%",
        }
      }
      testID="subTitle"
    >
      Up
    </Text>
  </View>
</View>
`;
